### PR TITLE
feat(internet-exposed): New public exposed checks

### LINKED ISSUE #3237: Improve publicly accessible checks to include targets of ELBs
### New feature motivation

I was looking at some prowler output, and at the "RDS instance not publicly accessible" check in particular
I think there are some improvements we can make.

Whilst the RDS instance itself might not be directly accessible from the internet, it might be behind an ELB with no auth which is sitting on the internet

So when it comes to checks for resources that live in a vpc that are "ensure X is not publicly accessible" we can make sure that there isn't an ELB for which the resource is part of a target group

### Solution Proposed

Enumerate ELBs, check for public ones, and then get their target groups
When doing checks for public resources within the other services, have some util function that can be called to check if the resource is a target for the ELB. It would also need to be able to handle IP addresses if the resource has an IP address associated to it (EKS node/ec2 instance/rds instance come to mind). 

Implement checks that see if the resource is a target for an elb (could be a new check, that is seperate to X_not_publicly_accessible - although based off of this check name it could form part of it. Alternatively make 2 checks: X_not_directly_publicly_accessible or X_not_publicly_accessible_via_elb, or something along those lines) 

One of the things I noticed is that awslambda_function_not_publicly_accessible only checks for IAM perms, not if its a target of an ALB.

Here are some of the types of resources which this type of check would apply to (I asked chatgpt what resources can be targets for ELBs. Im not sure if all of them will be applicable to what prowler is currently checking for, like the AWS Direct Connect, VPN, or peered VPCs)

* **EC2 Instances:** Elastic Load Balancers can distribute traffic to EC2 instances, which are virtual servers in Amazon's Elastic Compute Cloud (EC2).
* **ECS Tasks:** For applications running in Amazon Elastic Container Service (ECS), ELBs can route traffic to ECS tasks, which are Docker containers running on EC2 instances or AWS Fargate.
* **Lambda Functions:** Application Load Balancers (ALBs) can invoke AWS Lambda functions based on HTTP(S) requests, enabling serverless architectures.
* **IP Addresses:** ELBs can route traffic to specific IP addresses. This is useful for targeting resources that are hosted in on-premises data centers connected via AWS Direct Connect or a VPN, or in peered VPCs.
* **Containers:** For containerized applications, ELBs can directly route traffic to Docker containers managed by ECS or Kubernetes on AWS EKS.
* **Auto Scaling Groups:** ELBs work seamlessly with Auto Scaling Groups to adjust the amount of compute resources in response to incoming application traffic.
* **Amazon EC2 Spot Fleets:** ELBs can distribute traffic across Spot Fleet instances, which are collections of EC2 spot instances.

### Describe alternatives you've considered

None

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jfagoagas**
### Context

Thanks to @abant07 for this work! 👏 

Fixes https://github.com/prowler-cloud/prowler/issues/3237

### Description

New checks:
- awslambda_function_not_directly_publicly_accessible_via_elbv2
- ec2_instance_not_directly_publicly_accessible_via_elb
- ec2_instance_not_directly_publicly_accessible_via_elbv2

To-Do:
- [ ] Verify ALB/ELB and Instance/Lambda security group and ports
- [ ] Include the target groups in the ALB/ELB object

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/elbv2/elbv2_service.py [hunk 1]
```diff
@@ -51,6 +54,57 @@ def __describe_load_balancers__(self, regional_client):
...
+    def __describe_target_groups__(self, regional_client):
+        logger.info("ELBv2 - Describing target groups...")
+        try:
+            for lb in self.loadbalancersv2:
+                try:
+                    if (
+                        lb.scheme == "internet-facing"
+                        and lb.type == "application"
+                        and len(lb.security_groups) > 0
+                    ):
```
Comment (jfagoagas) [inline prowler/providers/aws/services/elbv2/elbv2_service.py]: I'd store all with a check of `public` if this condition is met.

File: prowler/providers/aws/services/ec2/ec2_instance_not_directly_publicly_accessible_via_elbv2/ec2_instance_not_directly_publicly_accessible_via_elbv2.metadata.json [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  "ResourceType": "AwsEc2Instance",
+  "Description": "Check for EC2 instances behind internet facing ALB/NLB/GLB.",
+  "Risk": "Exposing an EC2 to a ALB/NLB/GLB that is internet facing can lead to comprimisation",
+  "RelatedUrl": "",
+  "Remediation": {
+    "Code": {
+      "CLI": "",
+      "NativeIaC": "",
+      "Other": "",
+      "Terraform": ""
```
Comment (jfagoagas) [inline prowler/providers/aws/services/ec2/ec2_instance_not_directly_publicly_accessible_via_elbv2/ec2_instance_not_directly_publicly_accessible_via_elbv2.metadata.json:21]: Complete this.

File: prowler/providers/aws/services/ec2/ec2_instance_not_directly_publicly_accessible_via_elb/ec2_instance_not_directly_publicly_accessible_via_elb.metadata.json [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  "ResourceType": "AwsEc2Instance",
+  "Description": "Check for EC2 instances behind internet facing classic load balancers.",
+  "Risk": "Exposing an EC2 to a classic load balancer that is internet facing can lead to comprimisation",
+  "RelatedUrl": "",
+  "Remediation": {
+    "Code": {
+      "CLI": "",
+      "NativeIaC": "",
+      "Other": "",
+      "Terraform": ""
```
Comment (jfagoagas) [inline prowler/providers/aws/services/ec2/ec2_instance_not_directly_publicly_accessible_via_elb/ec2_instance_not_directly_publicly_accessible_via_elb.metadata.json:21]: Complete this.

File: prowler/providers/aws/services/awslambda/awslambda_function_not_directly_publicly_accessible_via_elbv2/awslambda_function_not_directly_publicly_accessible_via_elbv2.metadata.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  "Description": "Check if Lambda functions have public application load balancer ahead of them.",
+  "Risk": "Publicly accessible services could expose sensitive data to bad actors.",
+  "RelatedUrl": "https://docs.aws.amazon.com/lambda/latest/dg/access-control-resource-based.html",
+  "Remediation": {
+    "Code": {
+      "CLI": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/Lambda/function-exposed.html",
+      "NativeIaC": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/Lambda/function-exposed.html",
+      "Other": "",
+      "Terraform": "https://www.trendmicro.com/cloudoneconformity/knowledge-base/aws/Lambda/function-exposed.html"
+    },
```
Comment (jfagoagas) [inline prowler/providers/aws/services/awslambda/awslambda_function_not_directly_publicly_accessible_via_elbv2/awslambda_function_not_directly_publicly_accessible_via_elbv2.metadata.json]: Review this.

File: prowler/providers/aws/services/ec2/ec2_instance_not_directly_publicly_accessible_via_elb/ec2_instance_not_directly_publicly_accessible_via_elb.metadata.json (hunk 3 above)
Comment (MrCloudSec) [reply prowler/providers/aws/services/ec2/ec2_instance_not_directly_publicly_accessible_via_elb/ec2_instance_not_directly_publicly_accessible_via_elb.metadata.json:21]: There is no information

File: prowler/providers/aws/services/ec2/ec2_instance_not_directly_publicly_accessible_via_elbv2/ec2_instance_not_directly_publicly_accessible_via_elbv2.metadata.json (hunk 2 above)
Comment (MrCloudSec) [reply prowler/providers/aws/services/ec2/ec2_instance_not_directly_publicly_accessible_via_elbv2/ec2_instance_not_directly_publicly_accessible_via_elbv2.metadata.json:21]: There is no information

File: prowler/providers/aws/services/elbv2/elbv2_service.py (hunk 1 above)
Comment (MrCloudSec) [reply prowler/providers/aws/services/elbv2/elbv2_service.py]: Done

**Comment by jfagoagas** [comment]
Closed in favor of https://github.com/prowler-cloud/prowler/pull/3985

**Pull request closed without merging**